src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -483,6 +483,84 @@ public static void Options_JsonSerializerContext_DoesNotFallbackToReflection()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(unsupportedValue, options));
```

Comment:
Not sure if this should be a "TrimmingTest" (maybe ideally?) or a RemoteExecutor + `DisableDefaultReflection` RuntimeConfiguration test, but it would be good to have a test that added a source generated context to the options, and then tried to serialize a non-trivial object when the feature switch is set. Kind of a more "end to end still works" test.